legend-engine-xt-javaPlatformBinding-pure/src/main/resources/core_java_platform_binding/legendJavaPlatformBinding/planConventions/metaLibrary.pure

```diff
@@ -75,13 +77,76 @@ function <<access.private>> meta::pure::executionPlan::platformBinding::legendJa
+{
+   print(if($debug.debug,|$debug.space+'generateEnumerationIfValidJavaIdentifier : \''+$type.name->toOne()+'\'\n', |''));
+   let enumPkgAndName   = $conventions->className($type);
+  
+   ^meta::external::language::java::metamodel::Enumeration(
+      simpleName  = $enumPkgAndName.simpleName,
+      package     = $enumPkgAndName.package,
+      modifiers   = meta::external::language::java::metamodel::Modifier.Public,
+      enumeration = $enums->map(e| ^meta::external::language::java::metamodel::EnumerationEntry(name=$conventions->identifier($e->cast(@Enum).name))),
+      methods     = [
+         javaMethod('public', meta::external::language::java::factory::javaString(), 'getName', [], 'return this.name();'),
+         javaMethod(['public', 'static'], $enumPkgAndName, 'getEnumFromPureName', javaParam(javaString(), 'pureName'), $enumPkgAndName->j_invoke('valueOf', j_parameter(javaString(), 'pureName'), $enumPkgAndName)->j_return())
+      ]
+   );
+}
+
+function meta::pure::executionPlan::platformBinding::legendJava::library::meta::generateEnumerationIfInValidJavaIdentifier(type:meta::pure::metamodel::type::Type[1], conventions: Conventions[1], enums: Any[*], debug: DebugContext[1]): meta::external::language::java::metamodel::Class[1]
+{
+   print(if($debug.debug,|$debug.space+'generateEnumerationIfInvalidJavaIdentifier : \''+$type.name->toOne()+'\'\n', |''));
+   let enumPkgAndName     = $conventions->className($type);
+   let mapType             = javaMap(javaString(), $enumPkgAndName);
+   let pureName               = j_parameter(javaString(), 'pureName');
+   let pureNameToEnumMap     = javaField(['private', 'static', 'final'], $mapType, 'pureNameToEnumMap', 'initValues()');
+   let values              = j_variable($mapType, 'values');
    
+   let addValues           = $enums->size()->range()->map(index | $values->j_invoke('put',  [j_string($enums->at($index)->cast(@Enum).name), $enumPkgAndName->j_field('ENUM' + $index->toString())]));
+   let getEnumFromPureNameCode    = j_if($pureName->j_eq(j_null()),
+                                          javaNullPointerException()->j_new([j_string('pureName of Enum is null.')])->j_throw(),
+                                          j_if(
+                                                $enumPkgAndName->j_field($pureNameToEnumMap)->j_invoke('containsKey', $pureName),
+                                                $enumPkgAndName->j_field($pureNameToEnumMap)->j_invoke('get', $pureName)->j_return(),
+                                                javaIllegalArgumentException()->j_new([j_string('No Enum constant : ')->j_plus($pureName)])->j_throw()
+                                              )
+                                        );
+   let initCode            = j_block($values->j_declare(javaHashMap(javaString(), $enumPkgAndName)->j_new([]))->concatenate($addValues)->concatenate($values->j_return()));
```

Comment:
Can you move this up next to 'addValues' stmt

Reply:
sure.

Reply:
done.